### PR TITLE
Argparse cli

### PR DESCRIPTION
**Pull request by nainakhan123**

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
\ No newline at end of file
+def predict_yolo(path_to_weights, path_to_test):
+    model = YOLO(path_to_weights)
+    model.predict(source=path_to_test)
+    
+def main():
+    parser = argparse.ArgumentParser()  
+    parser.add_argument('path_to_weights')
+    parser.add_argument('path_to_test')
+    # parser.add_argument('saved_images', type=bool, default=True)
```
Comment (salmanmaq) [inline test.py:12]: We can remove such comments from the code

File: train.py [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
-model = YOLO('/home/sumbalkhan12/Test/barcode-detecion/runs/detect/train5/weights/best.pt')
-model.train(data="/home/sumbalkhan12/Test/barcode-detecion/large_dataset/custom.yaml", epochs=2)
-print('Successfull2') 
-model.val()
+def train_yolo(weights_path, data_config, epochs):
+    print('Starting training...')
+    model = YOLO(weights_path)
+    model.train(data=data_config, epochs=epochs)
+    print('Training completed.')
+
```
Comment (salmanmaq) [inline train.py:9]: We can add code for model validation here as well.

File: test.py [hunk 3]
```diff
@@ -1,5 +1,19 @@
 from ultralytics import YOLO
+import argparse
```
Comment (salmanmaq) [inline test.py:3]: 2 empty lines between imports and first function

File: test.py [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
 
-# /home/sumbalkhan12/Test/barcode-detecion/large_dataset/yolo_test/images
-model = YOLO("/home/sumbalkhan12/Test/barcode-detecion/runs/detect/train5/weights/best.pt")
-model.predict(source="test4.png", save=True)
\ No newline at end of file
+def predict_yolo(path_to_weights, path_to_test):
+    model = YOLO(path_to_weights)
+    model.predict(source=path_to_test)
+
```
Comment (salmanmaq) [inline test.py:7]: 2 empty lines between functions

**Comment by nainakhan123** [comment]
Thank you for your feedback. I have made the changes please review the code again.

**Comment by salmanmaq** [comment]
> Empty lines, extra comments removed

You can add 1 more empty line for functions. The convention is 2 empty lines.
Additionally you can add the validation at the end of the training function

**Comment by nainakhan123** [comment]
I have made the required changes.

File: test.py [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+    parser = argparse.ArgumentParser()  
+    parser.add_argument('path_to_weights')
+    parser.add_argument('path_to_test')
+    
+
+    args = parser.parse_args()
+
+
+    predict_yolo(args.path_to_weights, args.path_to_test)
+
```
Comment (salmanmaq) [inline test.py:20]: We can add another empty line here

File: train.py [hunk 6]
```diff
@@ -1,6 +1,25 @@
...
+    parser = argparse.ArgumentParser(description='Train YOLOv8 on custom data')
+    parser.add_argument('weights', required=True)
+    parser.add_argument('data-config', required=True)
+    parser.add_argument('epochs', type=int, default=2)
+    
+
+    args = parser.parse_args()
+    
+    train_yolo(args.weights, args.data_config, args.epochs)
+
```
Comment (salmanmaq) [inline train.py:23]: We can add another empty line here as well

File: train.py [hunk 7]
```diff
@@ -1,6 +1,25 @@
...
+    print('Training completed.')
+    model.val()
+   
+
+def main():
+    parser = argparse.ArgumentParser(description='Train YOLOv8 on custom data')
+    parser.add_argument('weights', required=True)
+    parser.add_argument('data-config', required=True)
+    parser.add_argument('epochs', type=int, default=2)
+
```
Comment (salmanmaq) [inline train.py:18]: Can remove extra empty line here

File: test.py [hunk 8]
```diff
@@ -1,5 +1,22 @@
...
+def predict_yolo(path_to_weights, path_to_test):
+    model = YOLO(path_to_weights)
+    model.predict(source=path_to_test)
+
+
+def main():
+    parser = argparse.ArgumentParser()  
+    parser.add_argument('path_to_weights')
+    parser.add_argument('path_to_test')
+
```
Comment (salmanmaq) [inline test.py:14]: Can remove extra empty line here

File: test.py [hunk 9]
```diff
@@ -1,5 +1,22 @@
...
+def main():
+    parser = argparse.ArgumentParser()  
+    parser.add_argument('path_to_weights')
+    parser.add_argument('path_to_test')
+    
+
+    args = parser.parse_args()
+
+
```
Comment (salmanmaq) [inline test.py:18]: Can remove extra empty line here

**Pull request merged**